lib/types.ts

```diff
@@ -284,6 +284,13 @@ export type NonEmptyObject<T extends {}> = keyof T extends never ? never : T;
 /** Merge 2 types, properties types from the latter override the ones defined on the former type */
 export type Merge<M, N> = Omit<M, keyof N> & N;
 
+type _MergeN<T extends readonly any[], Result> = T extends readonly [infer Head, ...(infer Tail)]
```

Comment:
Ah, that's only available since TypeScript@4.0 – https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html